molgenis-core-ui/src/main/java/org/molgenis/core/ui/admin/usermanager/UserManagerServiceImpl.java

```diff
@@ -50,6 +54,20 @@ public long getActiveSessionsCount() {
     return securityContextRegistry.getSecurityContexts().count();
   }
 
+  @Override
+  @PreAuthorize("hasAnyRole('ROLE_SU','ROLE_MANAGER')")
+  public List<String> getActiveSessionUserNames() {
+    return securityContextRegistry
+        .getSecurityContexts()
+        .map(SecurityContext::getAuthentication)
+        .filter(Objects::nonNull)
+        .map(Authentication::getPrincipal)
+        .filter(
+            principal -> principal instanceof org.springframework.security.core.userdetails.User)
```

Comment:
I think principal.getName is probably better, it's not safe to assume too much about the Authentication object

Reply:
But what if principal is of different type but still has a getName ? , then the type check in the PR seems saver to me, as its skips it in case of unexpected type

Reply:
getPrincipal returns a Object so there is no guarantee you get de a Principal with a getName

Reply:
Authentication is a principal, so you can call getName on it. They've done their best already to make sure they return the correct name.

Reply:
Updated the PR as suggested, seems to work :). But whole thing depends on us using the Principal.name as userName. But i guess that kind of true for all implementations where we join the list of users with the list of authentications